extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -873,17 +879,29 @@ private void checkForDurableStorageConnectorImpl()
     }
   }
 
-  private <T> T contactOverlord(final ListenableFuture<T> future)
+  private <T> T contactOverlord(final ListenableFuture<T> future, String queryId)
   {
     try {
       return FutureUtils.getUnchecked(future, true);
     }
     catch (RuntimeException e) {
+      if (e.getCause() instanceof HttpResponseException) {
+        HttpResponseException httpResponseException = (HttpResponseException) e.getCause();
```

Comment:
We should attach `e.getCause()` in the NotFound exception that we are throwing.

Reply:
I think its ok since we are returning the error for a specific response code. We could log an info here though.

Reply:
We may not need to log anything since this is bad input from the user and it will muddy the logs.

Reply:
yes. except when user says its not bad input and we can't debug or prove because there is nothing in logs :)